### PR TITLE
Lite: setting extras & fix CI

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda @justusschock @awaelchli

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
Ooops, we accidentally imported `from pytorch_lightning.accelerators import TPUAccelerator` instead of from Lite:

```

___________ ERROR collecting tests/tests_lite/strategies/test_xla.py ___________
strategies/test_xla.py:28: in <module>
    from pytorch_lightning.accelerators import TPUAccelerator
../../src/pytorch_lightning/__init__.py:35: in <module>
    from pytorch_lightning.callbacks import Callback  # noqa: E402
../../src/pytorch_lightning/callbacks/__init__.py:[22](https://github.com/Lightning-AI/lightning/actions/runs/3281889739/jobs/5404532607#step:11:23): in <module>
    from pytorch_lightning.callbacks.lr_finder import LearningRateFinder
../../src/pytorch_lightning/callbacks/lr_finder.py:[24](https://github.com/Lightning-AI/lightning/actions/runs/3281889739/jobs/5404532607#step:11:25): in <module>
    from pytorch_lightning.tuner.lr_finder import _LRFinder, lr_find
../../src/pytorch_lightning/tuner/lr_finder.py:38: in <module>
    from tqdm import tqdm
E   ModuleNotFoundError: No module named 'tqdm'
```

File: .github/workflows/ci-lite-test-full.yml [hunk 1]
```diff
@@ -75,27 +79,28 @@ jobs:
...
+      run: python -m pytest lightning_lite
+
     - name: Testing Warnings
-      # the stacklevel can only be set on >=3.7
+      # the stack-level can only be set on >=3.7
       if: matrix.python-version != '3.7'
-      working-directory: tests/tests_lite
-      # needs to run outside of `pytest`
+      working-directory: ./tests/tests_lite
+      # needs to run outside `pytest`
```
Comment (carmocca) [inline .github/workflows/ci-lite-test-full.yml:97]: `of` sounds clearer to me, not sure if removing it is correct.

File: .github/workflows/ci-lite-test-full.yml [hunk 2]
```diff
@@ -75,27 +79,28 @@ jobs:
...
       run: python utilities/test_warnings.py
 
     - name: Testing Lite
-      working-directory: tests/tests_lite
+      working-directory: ./tests
```
Comment (carmocca) [inline .github/workflows/ci-lite-test-full.yml:101]: These cwd changes are inconsistent with the other CI files.

Are they necessary? I suggest undoing them to avoid being inconsistent

File: .github/workflows/ci-lite-test-full.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/ci-lite-test-full.yml:97]:
```suggestion
      # needs to run outside of `pytest`
```

File: .github/workflows/ci-lite-test-full.yml (hunk 2 above)
Comment (Borda) [reply .github/workflows/ci-lite-test-full.yml:101]: tbh, I would rather use this to have generated files in tests folder but I ma also fine to revert it... :otter:

Comment (carmocca) [reply .github/workflows/ci-lite-test-full.yml:101]: Im fine with having them, but on a separate PR that does it for all workflows

File: .github/workflows/ci-lite-test-full.yml [hunk 3]
```diff
@@ -106,19 +111,18 @@ jobs:
...
       if: success()
-      working-directory: tests/tests_lite
+      working-directory: ./tests
       run: |
         coverage report
         coverage xml
 
     - name: Upload coverage to Codecov
       uses: codecov/codecov-action@v3
-      if: always()
```
Comment (carmocca) [inline .github/workflows/ci-lite-test-full.yml:116]: Is this removed on purpose, why?

Again, I suggest these coverage related changes are done in a separate PR for all workflows.

Comment (Borda) [reply .github/workflows/ci-lite-test-full.yml:116]: yes, because if pytest/workflow fails, we shall not run this step because there is not report... and if you check it is not in other workflows

File: .github/workflows/ci-lite-test-full.yml (hunk 2 above)
Comment (Borda) [reply .github/workflows/ci-lite-test-full.yml]: @carmocca reverted :rabbit:

**Review by carmocca — APPROVED** [review]

File: tests/tests_lite/utilities/test_distributed.py [hunk 4]
```diff
@@ -1,9 +1,29 @@
+from functools import partial
+
 import pytest
 import torch
-from tests_lite.helpers.runif import RunIf
 
+from lightning_lite.plugins.environments import LightningEnvironment
 from lightning_lite.utilities.distributed import gather_all_tensors
-from tests_pytorch.core.test_results import spawn_launch
+from pytorch_lightning.accelerators import CPUAccelerator, CUDAAccelerator, MPSAccelerator
```
Comment (Borda) [inline tests/tests_lite/utilities/test_distributed.py:8]: is this intentional?

Comment (carmocca) [reply tests/tests_lite/utilities/test_distributed.py:9]: Yes. We don't really want `tests_lite` to import from `tests_pytorch`. It was done out of laziness. This also explains why you had to install torchmetrics, tensorboard, etc

File: .azure/gpu-tests-pytorch.yml [hunk 5]
```diff
@@ -134,8 +134,8 @@ jobs:
     - bash: bash .actions/pull_legacy_checkpoints.sh
       displayName: 'Get legacy checkpoints'
 
-    - bash: python -m coverage run --source pytorch_lightning -m pytest
-      workingDirectory: src/pytorch_lightning
+    - bash: python -m pytest pytorch_lightning
```
[the remark this answers is not in the document]
Comment (carmocca) [reply .azure/gpu-tests-pytorch.yml:137]: These are doctests

**Review by manskx — APPROVED** [review]

File: .azure/gpu-tests-pytorch.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (Borda) [reply .azure/gpu-tests-pytorch.yml:137]: And I think it would be overwritten by the next testing anyway... so in this case, we would need to upload it in the mean time...
